File: pyramid-generator.py
def pyramid(pattern, rows, inverted):
    lines = []

    for i in range(rows):
        row_num = rows - i if inverted else i + 1

        space_count = rows - row_num
        pattern_count = 2 * row_num - 1

        line = " " * space_count + pattern * pattern_count
        lines.append(line)

    return "\n" + "\n".join(lines) + "\n"

#ex

print("o"*4)
print(" "*3)
print("ABC"*2)

File: pyramid-generator.js
function pyramid(pattern, rows, inverted) {
  const lines = [];

  for (let i = 0; i < rows; i++) {
    const rowNum = inverted ? rows - i : i + 1;

    const spaceCount = rows - rowNum;
    const patternCount = 2 * rowNum - 1;

    const line = " ".repeat(spaceCount) + pattern.repeat(patternCount);
    lines.push(line);
  }

  return "\n" + lines.join("\n") + "\n";
}

//

"".repeat();

console.log("o".repeat(3));
console.log(" ".repeat(4));
console.log("ABC".repeat(2));
console.log("x".repeat(0));
console.log("a".repeat(2.8));